charts/spire/templates/_spire-lib.tpl

```diff
@@ -50,3 +50,15 @@
 {{- printf "%s/%s" $registry .image.repository -}}
 {{- end -}}
 {{- end }}
+
+{{/*
+Take in an array of, '.', a failure string to display, and boolean to to display it,
+if strict checking is enabled and the boolean is true
+*/}}
+{{- define "spire-lib.strict-check" }}
+{{- if (dig "spire" "strictChecking" false (index . 0).Values.global) }}
+{{- if (index . 2) }}
+{{- fail (index . 1) }}
```

Comment:
Yeah... A usage documentation item would be good. Will add.

Ordering of the yaml doesn't matter. useage is like:
```
{{- include "spire-lib.strict-check" (list . "trustDomain must be set" (eq (include "spire-lib.trust-domain" .) "example.org"))}}
```

where the arguments are specified in a list so wont be reordered, but takes up less space then doing it with dictionary passing.

args are root, an error to display, and a boolean on if to fail and display when true

Reply:
actually, the arguments are kind of described here:
https://github.com/spiffe/helm-charts/pull/229/files#diff-789477a4f2a924ca09b7f80a469fc8827159a3d3d8a6281c37018dc16b60c094R55-R56

There any particular wording changes that might make it more understandable?

Reply:
> * What is `index . 2`?
> * What is `index . 1`?

Oh. Maybe I'm not understanding the question either....
```index thing 1``` is what might be done in other languages like ```thing[1]``` or sometimes ```thing.1```. its looking up the second argument passed to the macro in a list. (0 based indexing)

Reply:
I'm also having some difficulty processing this one

```
{{- include "spire-lib.strict-check" (list . "trustDomain must be set" (eq (include "spire-lib.trust-domain" .) "example.org"))}}
```

I think even though this might be less syntax a dict could make it more descriptive and understandable. After reading this example 3 times I'm still not sure I get it.

Reply:
Ah got it now. Probably a good suggestion to assign them to variables that makes it more understanable indeed.

I would suggest following names.

```
{{ $root := index 0 }}
{{ $message := index 1 }}
{{ $condition := index 2 }}
```

Reply:
Done